src/keyboard/mod.rs

```diff
@@ -0,0 +1,258 @@
+//! Utilities for keyboard handling
+//!
+//! This module provides utilities for keyboardand keymap handling: keymap interpretation
+//! and forwarding keystrokes to clients using xkbcommon.
+//!
+//! You can first create a `KbdHandle` using the `create_keyboard_handler` function in this module.
+//! The handle you obtained can be cloned to access this keyboard state from different places. It is
+//! expected that such a context is created for each keyboard the compositor has access to.
```

Comment:
This also means libinput does not generate seat-events. I will have to track seats with each device-event. The library really seems to expect you to handle devices instead of seats like the wayland protocol.